src/libstd/sys/hermit/thread.rs

```diff
@@ -49,26 +48,32 @@ impl Thread {
         p: Box<dyn FnOnce()>,
         core_id: isize,
     ) -> io::Result<Thread> {
-        let p = box p;
+        let p = Box::into_raw(box p);
         let mut tid: Tid = u32::MAX;
         let ret = abi::spawn(
             &mut tid as *mut Tid,
             thread_start,
-            &*p as *const _ as *const u8 as usize,
+            p as *mut u8 as usize,
```

Comment:
why is this intermediate `as *mut u8` still necessary? Can't any narrow raw pointer be cast to `usize` directly?

Reply:
It is not necessary. Thanks for pointing out!